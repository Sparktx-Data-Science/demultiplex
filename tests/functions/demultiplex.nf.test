nextflow_function {

    name "Test Functions"
    script "workflows/demultiplex.nf"

    // TODO test("Test Function summary_params = NfcoreSchema.paramsSummaryMap")

    test("Test Function extract_csv") {

        function "extract_csv"

        when {
            function {
                """
                input[0] = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                """
            }
        }

        then {
            // FIXME
            assert function.failed
            assert snapshot(function.result).match()
        }

    }

    // TODO test("Test Function all_columns = input_schema.columns.keySet().collect")
    // TODO test("Test Function commons = all_columns.intersect")
    // TODO test("Test Function diffs = all_columns.plus")
    // TODO test("Test Function parse_flowcell_csv")
    // TODO test("Test Function flowcell        = file")
    // TODO test("Test Function samplesheet     = file")

}
