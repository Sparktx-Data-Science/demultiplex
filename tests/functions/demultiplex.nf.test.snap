{
    "Test Function extract_csv": {
        "content": null,
        "timestamp": "2023-06-05T10:39:32+0000"
    }
}